astropy/cosmology/funcs.py

```diff
@@ -47,46 +79,19 @@ def z_at_value(func, fval, zmin=1e-8, zmax=1000, ztol=1e-8, maxfun=500):
     maxfun : int, optional
        The maximum number of function evaluations allowed in the
        optimization routine (default 500).
+    nbins : float, optional
+        If passing an array of ``fval``, this specifies the number
+        of gridpoints to use for interpolation.
+    logspace: bool, optional
+        If passing an array of ``fval``, this specifies whether
+        to create the gridpoints in logarithmic space
 
     Returns
     -------
-    z : float
+    z : float or array_like
       The redshift ``z`` satisfying ``zmin < z < zmax`` and ``func(z) =
       fval`` within ``ztol``.
 
-    Notes
```

Comment:
Looks like this whole section was removed without any replacement text.

Reply:
The `Notes` section here was advice for how to operate on arrays using interpolation. Since the function does this automatically now, I don't see a need for this docstring.

Reply:
I do wonder why, given that the docs already had an explanation on how to do this, that the functionality wasn't incorporated into the method to begin with.

Reply:
I vaguely recall that the issue was that in normal usage you basically iterate to a pretty exact solution, while with interpolation it depends on your number of bins, etc. I.e., a case perhaps where the wish for perfection stopped the "good enough".

But this does bring up a more general issue, whether the result of the call should depend on whether something is passed in as an array or as a scalar.  There is an argument to be made to create a more general interpolation method and just refer to that in the docstring. But, then again, I may be aiming for too much perfection...

p.s.  Sorry, right now time just to chime in, not for a proper review.
p.s. @MilesCranmer: thanks for the PR!

Reply:
Thanks!
Regarding different answers between scalar/array, I could add a warning for this? I also agree: there's probably no perfect solution here.

Reply:
Actually, instead I think your idea to put it in the docstring is a better idea than a warning - I can add that!

Reply:
Done - added a note to the docstring.